docs/features/kubernetes/proxy.md

```diff
@@ -0,0 +1,87 @@
+---
+id: proxy
+title: Kubernetes Backend Proxy Endpoint
+sidebar_label: Proxy
+description: Interacting with the Kubernetes API in Backstage plugins
+---
+
+[Contributors](https://backstage.io/docs/overview/glossary#backstage-user-profiles) wanting to
+create developer portal experiences based on data from Kubernetes (e.g. for
+interacting with [Custom
+Resources](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)
+beyond the default behaviors of the existing Kubernetes plugin) can leverage the
+Kubernetes backend plugin's proxy endpoint to allow them to make arbitrary
+requests to the [REST
+API](https://kubernetes.io/docs/reference/using-api/api-concepts/).
+
+Here is a snippet fetching namespaces from a cluster configured with the
+`google` [auth provider](https://backstage.io/docs/features/kubernetes/configuration#clustersauthprovider):
+
+```typescript
+import {
+  discoveryApiRef,
+  googleAuthApiRef,
+  useApi,
+} from '@backstage/core-plugin-api';
+
+const CLUSTER_NAME = ''; // use a known cluster name
+
+// get a bearer token from Google
+const googleAuthApi = useApi(googleAuthApiRef);
+const token = await googleAuthApi.getAccessToken(
+  'https://www.googleapis.com/auth/cloud-platform',
+);
+
+const discoveryApi = useApi(discoveryApiRef);
+const kubernetesBaseUrl = await discoveryApi.getBaseUrl('kubernetes');
+const kubernetesProxyEndpoint = `${kubernetesBaseUrl}/proxy`;
+
+// fetch namespaces
+await fetch(`${kubernetesProxyEndpoint}/api/v1/namespaces`, {
+  method: 'GET',
+  headers: {
+    'X-Kubernetes-Cluster': CLUSTER_NAME,
+    Authorization: `Bearer ${token}`,
+  },
+});
+```
+
+## How it works
+
+The proxy will interpret the
+[`X-Kubernetes-Cluster`
```

Comment:
I mentioned this separately in a another PR, but the `X-` prefix is discouraged and we should try to remove it if possible. Ofc not as part of this PR but just wanna get the info out there 😅